src/prefect/engine/task_runner.py

```diff
@@ -873,9 +874,16 @@ def get_task_run_state(
                 timeout_handler or prefect.utilities.executors.timeout_handler
             )
             raw_inputs = {k: r.value for k, r in inputs.items()}
-            result = timeout_handler(
-                self.task.run, timeout=self.task.timeout, **raw_inputs
-            )
+
+            if self.task.log_stdout:
+                with redirect_stdout(prefect.utilities.logging.log_stdout):  # type: ignore
```

Comment:
I think we should log with the Task Runner logger here (`self.logger`) to keep the logs in an intuitive place; maybe you can take your `RedirectToLog` and allow it to be initialized with a logger?  E.g., so this would be
```python
with redirect_stdout(RedirectToLog(self.logger)):
```